homeassistant/components/cloud/http_api.py

```diff
@@ -81,39 +90,133 @@ class CloudLogoutView(HomeAssistantView):
     name = 'api:cloud:logout'
 
     @asyncio.coroutine
+    @_handle_cloud_errors
     def post(self, request):
-        """Validate config and return results."""
+        """Handle logout request."""
         hass = request.app['hass']
-        try:
-            with async_timeout.timeout(REQUEST_TIMEOUT, loop=hass.loop):
-                yield from \
-                    hass.data[DOMAIN]['cloud'].async_revoke_access_token()
+        auth = hass.data['cloud']['auth']
 
-            hass.data[DOMAIN].pop('cloud')
+        with async_timeout.timeout(REQUEST_TIMEOUT, loop=hass.loop):
+            yield from hass.async_add_job(auth.logout)
 
-            return self.json({
-                'result': 'ok',
-            })
-        except asyncio.TimeoutError:
-            return self.json_message("Could not reach the server.", 502)
-        except cloud_api.UnknownError as err:
-            return self.json_message(
-                "Error communicating with the server ({}).".format(err.status),
-                502)
+        return self.json_message('ok')
 
 
 class CloudAccountView(HomeAssistantView):
-    """Log out of the Home Assistant cloud."""
+    """View to retrieve account info."""
 
     url = '/api/cloud/account'
     name = 'api:cloud:account'
 
     @asyncio.coroutine
     def get(self, request):
-        """Validate config and return results."""
+        """Get account info."""
         hass = request.app['hass']
+        auth = hass.data['cloud']['auth']
 
-        if 'cloud' not in hass.data[DOMAIN]:
+        if not auth.is_logged_in:
             return self.json_message('Not logged in', 400)
 
-        return self.json(hass.data[DOMAIN]['cloud'].account)
+        return self.json(_auth_data(auth))
+
+
+class CloudRegisterView(HomeAssistantView):
+    """Register on the Home Assistant cloud."""
+
+    url = '/api/cloud/register'
+    name = 'api:cloud:register'
+
+    @asyncio.coroutine
+    @_handle_cloud_errors
+    @RequestDataValidator(vol.Schema({
+        vol.Required('email'): str,
+        vol.Required('password'): vol.All(str, vol.Length(min=6)),
+    }))
+    def post(self, request, data):
+        """Handle registration request."""
+        hass = request.app['hass']
+
+        with async_timeout.timeout(REQUEST_TIMEOUT, loop=hass.loop):
+            yield from hass.async_add_job(
+                auth_api.register, hass, data['email'], data['password'])
+
+        return self.json_message('ok')
+
+
+class CloudConfirmRegisterView(HomeAssistantView):
+    """Confirm registration on the Home Assistant cloud."""
+
+    url = '/api/cloud/confirm_register'
+    name = 'api:cloud:confirm_register'
+
+    @asyncio.coroutine
+    @_handle_cloud_errors
+    @RequestDataValidator(vol.Schema({
```

Comment:
I think we should defined the Schema not inline. Maybe in `cloud/const.py` or `cloud/schema.py`